### DISCUSSION TITLE
get state in a nextjs server component with new app router

### DISCUSSION THREAD

**Discussion by laclance** [pmndrs/zustand#1907] [Q&A] [answered]
Is it possible to get state inside a nextjs server component with the new app router, i.e render a server component and get state with useStore().getState() or some other method?

**Comment by dai-shi** [comment, marked as answer]
It's basically impossible to read client state on server. I'm interested in creating such a library that somehow supports it, but it will take time.

**Reply by laclance** [reply]
OK thanks for the reply.
